src/snapshot/patch_compute.rs

```diff
@@ -10,16 +10,27 @@ use super::{
     InstanceSnapshot, InstanceWithMeta, RojoTree,
 };
 
-pub fn compute_patch_set(snapshot: &InstanceSnapshot, tree: &RojoTree, id: Ref) -> PatchSet {
+pub fn compute_patch_set(
+    snapshot: &Option<InstanceSnapshot>,
+    tree: &RojoTree,
+    id: Ref,
+) -> PatchSet {
```

Comment:
I like this change!

Good opportunity for a Rust note: always prefer `Option<&T>` over `&Option<T>`. They're almost exactly the same, but `Option<&T>` is more flexible because it allows creating an Option of something you don't own. For other code, it just results in the ampersand moving a couple characters over.

Reply:
Super interesting, thank you so much!